app/lib/dates.lib.js

```diff
@@ -9,6 +9,27 @@ const february = 2
 const lastDayOfFebruary = 28
 const lastDayOfFebruaryLeapYear = 29
 
+/**
+ * Given a return version and return cycle end date, return the earliest date as a string in the ISO format.
+ *
+ * @param {Array} returnVersion - The list of dates to check.
+ * @returns {String} - The earliest date as a string in the ISO format.
+ * @private
+ */
+function earliestDate(dates) {
+  const _dates = [...dates]
+    .filter((date) => {
+      return date
+    })
+    .map((date) => {
+      return new Date(date)
+    })
+
+  const _earliestDate = new Date(Math.min(..._dates))
+
+  return _earliestDate
```

Comment:
If we expect `dates` to be an array, we can work with it directly.

I think we can also split up the steps rather than do them in one pass. IMHO, I think it makes it clearer what is happening.

We can also directly return the result rather than temporarily instantiating a value.

Finally, having then looked at the tests, we hadn't covered what happens if the array we pass in is empty or contains all null values. The result was unexpected, which led me to add in specific handling for that scenario.

```suggestion
  const allEmptyValuesRemoved = dates.filter((date) => {
    return date
  })
  
  if (allEmptyValuesRemoved.length === 0) {
    throw Error('No dates provided to determine earliest')
  }

  const valuesAsDates = allEmptyValuesRemoved.map((date) => {
    return new Date(date)
  })

  const earliestDateTimestamp = Math.min(...valuesAsDates)

  return new Date(earliestDateTimestamp)
```

P.S. An `_` is a convention for marking a _function_ as private, i.e., it won't be exported from the module and is only intended for internal use. You don't typically prefix variables with `_`.